### ISSUE TITLE
Expression browser doesn't handle empty responses well.

### ISSUE THREAD

**Issue by noony** [prometheus/prometheus#3515]
**What did you do?**
I'm testing rules before integration in alert manager .rules.yml. So I'm running this querying on prometheus with and hour.
````
(hour() >= bool 1) and on (instance) (up == 1)
````

**What did you expect to see?**
the list of targets with up == 1 because there is targets with up == 1 and hour() is > to 1

**What did you see instead? Under which circumstances?**
Javascript error in console :
````
Uncaught TypeError: Cannot read property 'length' of null
    at Prometheus.Graph.handleConsoleResponse (graph.js?v=0a74f98628a0463dddc90528220c94de5032d1a0:723)
    at success (graph.js?v=0a74f98628a0463dddc90528220c94de5032d1a0:418)
    at Object.success (graph.js?v=0a74f98628a0463dddc90528220c94de5032d1a0:432)
    at i (jquery.min.js?v=0a74f98628a0463dddc90528220c94de5032d1a0:2)
    at Object.fireWith [as resolveWith] (jquery.min.js?v=0a74f98628a0463dddc90528220c94de5032d1a0:2)
    at A (jquery.min.js?v=0a74f98628a0463dddc90528220c94de5032d1a0:4)
    at XMLHttpRequest.<anonymous> (jquery.min.js?v=0a74f98628a0463dddc90528220c94de5032d1a0:4)
````

Json return by the query string `/api/v1/query?query=(hour()%20>%3D%20bool%201)%20and%20on%20(instance)%20(up%20%3D%3D%201)&time=1511781829.405&_=1511781184970`
````
{"status":"success","data":{"resultType":"vector","result":null}}
````

**Environment**

* System information:

	Linux 4.4.0-96-generic x86_64

* Prometheus version:

````
prometheus, version 2.0.0 (branch: HEAD, revision: 0a74f98628a0463dddc90528220c94de5032d1a0)
  build user:       root@615b82cb36b6
  build date:       20171108-07:11:59
  go version:       go1.9.2
````

**Comment by brian-brazil** [comment]
That's a bug in the graph code, but the empty response is correct.

**Comment by roidelapluie** [comment]
Fix proposed in #3521
